libdnf/utils/os-release.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Licensed under the GNU Lesser General Public License Version 2.1
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "os-release.hpp"
```

Comment:
Can we rename this file to something else? the handling of os-release is only one of the functions, and we should use these functions in other parts of libdnf (for example, to get rid of the duplicate arch table in `dnf-context.cpp` and in the Python code).

And I fully expect more functions to arrive later...

Reply:
Agreed. There actually is work in progress by @jrohel that addresses architecture detection in a broader sense (mostly to consolidate the logic to one single place), but it's not ready yet, and I needed this functionality right now, so I figured we could add it temporarily this way and factor out later.

I talked to Jarek today and we decided that I would create the new file `utils/arch.cpp` and move the detection routine there (and the os-release functions to `utils/utils.[hc]pp`) for now. Once he's ready with his work, it would replace the contents of `utils/arch.[hc]pp`. I update this PR later today, together with the other additions from my local checkout that I'm working on.